testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -389,55 +389,76 @@ public void testTrinoCreatesSparkDrops()
 
-        Row row1 = row("b", new byte[] {15, -15, 2, -16, -2, -2}, 1002);
-        String selectByString = "SELECT * FROM %s WHERE _string = 'b'";
+        Row row1 = row("b", new byte[]{15, -15, 2, -16, -2, -2}, 1002, 2);
+        String selectByString = "SELECT _string, _varbinary, _bigint, _struct._field FROM %s WHERE _string = 'b'";
         assertThat(onTrino().executeQuery(format(selectByString, trinoTableName)))
                 .containsOnly(row1);
+        // TODO
+        // Parquet: [b, [15, -15, 2, -16, -2, -2], 1002, null]
+        // ORC: Caused by: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 50.0 failed 1 times, most recent failure: Lost task 0.0 in stage 50.0 (TID 86) (spark executor driver): java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 1
+        //  at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
+        //  at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
+        //  at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
         assertThat(onSpark().executeQuery(format(selectByString, sparkTableName)))
                 .containsOnly(row1);
 
-        Row row2 = row("a", new byte[] {15, -15, 2, -16, -2, -1}, 1001);
-        String selectByVarbinary = "SELECT * FROM %s WHERE _varbinary = X'0ff102f0feff'";
+        Row row2 = row("a", new byte[]{15, -15, 2, -16, -2, -1}, 1001, 1);
+        String selectByVarbinary = "SELECT _string, _varbinary, _bigint, _struct._field FROM %s WHERE _varbinary = X'0ff102f0feff'";
         assertThat(onTrino().executeQuery(format(selectByVarbinary, trinoTableName)))
                 .containsOnly(row2);
         assertThat(onSpark().executeQuery(format(selectByVarbinary, sparkTableName)))
                 .containsOnly(row2);
 
-        onTrino().executeQuery("DROP TABLE " + trinoTableName);
+        Row row3 = row("a", new byte[]{15, -15, 2, -16, -2, -1}, 1001, 1);
+        String selectNested = "SELECT _string, _varbinary, _bigint, _struct._field FROM %s WHERE _struct._field = 1";
+        assertThat(onTrino().executeQuery(format(selectNested, trinoTableName)))
+                .containsOnly(row3);
+        assertThat(onSpark().executeQuery(format(selectNested, sparkTableName)))
+                .containsOnly(row3);
     }
 
     @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS, ICEBERG_REST, ICEBERG_JDBC}, dataProvider = "storageFormatsWithSpecVersion")
     public void testTrinoReadsSparkPartitionedTable(StorageFormat storageFormat, int specVersion)
     {
+        if (storageFormat == StorageFormat.ORC) {
+            throw new SkipException("Open Spark issue https://github.com/apache/iceberg/issues/3139 to read ORC table with nested partition column.");
```

Comment:
Instead of skipping , we should rather check that Trino does perform as expected and Spark throws an exception.